CMakeLists.txt

```diff
@@ -143,71 +113,59 @@ add_library(jluna SHARED
     include/gc_sentinel.hpp
 )
 
-set_target_properties(jluna PROPERTIES
-    LINKER_LANGUAGE C
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
-target_compile_features(jluna PUBLIC cxx_std_20)
-
-### C ADAPTER ##
-
-add_library(jluna_c_adapter SHARED
-    .src/c_adapter.hpp
-    .src/c_adapter.cpp
-)
-
-target_include_directories(jluna_c_adapter
-    PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>
-    PUBLIC $<BUILD_INTERFACE:${JULIA_INCLUDE}>
-)
-
-target_link_libraries(jluna_c_adapter PUBLIC $<BUILD_INTERFACE:${JULIA_LIB}>)
-
-set_target_properties(jluna_c_adapter PROPERTIES
-    LINKER_LANGUAGE C
-    LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
-
 target_link_libraries(jluna PUBLIC jluna_c_adapter)
+target_include_directories(jluna PRIVATE "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>")
 
-### EXECUTABLES ###
-
-add_executable(JLUNA_TEST
-    .test/main.cpp
-    .test/test.hpp
-)
-target_link_libraries(JLUNA_TEST jluna ${JULIA_LIB})
+### HACK: export all symbols on Windows ###
 
-add_executable(JLUNA_BENCHMARK
-    .benchmark/main.cpp
-    .benchmark/benchmark.hpp
-    .benchmark/benchmark_aux.hpp
+set_target_properties(jluna jluna_c_adapter PROPERTIES
+    WINDOWS_EXPORT_ALL_SYMBOLS YES
+    LIBRARY_OUTPUT_DIRECTORY ${C_ADAPTER_PATH}
 )
-target_link_libraries(JLUNA_BENCHMARK jluna ${JULIA_LIB})
 
-### INSTALL ###
+### Install rules ###
 
-file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/jluna-config.cmake [[
-if (NOT ((CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION STRGREATER_EQUAL "12.0.0") OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION STRGREATER_EQUAL "10.0.0")))
-    message(FATAL_ERROR "Currently, the only supported compilers are G++10, G++11 (recommended) and Clang-12")
+if(NOT CMAKE_SKIP_INSTALL_RULES)
+    include(cmake/install-rules.cmake)
 endif()
-include("${CMAKE_CURRENT_LIST_DIR}/jluna-targets.cmake")
-]])
 
-install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/jluna-config.cmake
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/jluna
-)
+### Developer mode ###
+
+if(NOT jluna_DEVELOPER_MODE)
+    return()
+elseif(NOT PROJECT_IS_TOP_LEVEL)
+    message(
+        AUTHOR_WARNING
+        "Developer mode is intended for developers of jluna"
+    )
+endif()
 
-install(TARGETS jluna jluna_c_adapter EXPORT jluna_targets)
+include(CTest)
+if(jluna_DEVELOPER_MODE)
 
-install(EXPORT jluna_targets
-    FILE jluna-targets.cmake
-    NAMESPACE jluna::
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/jluna
-)
+    ### Declare Test ###
 
-install(FILES jluna.hpp ${headers} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/jluna)
-install(DIRECTORY include .src DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/jluna)
+    option(BUILD_TESTING "Enable jluna_test" ON)
```

Comment:
This variable is already created by [CTest](https://cmake.org/cmake/help/latest/module/CTest.html):

> The module automatically creates a `BUILD_TESTING` option that selects whether to enable testing support (`ON` by default).